packages/components/select/src/useSelect.ts

```diff
@@ -464,7 +464,7 @@ export const useSelect = (props, states: States, ctx) => {
     for (let i = states.cachedOptions.size - 1; i >= 0; i--) {
       const cachedOption = cachedOptionsArray.value[i]
       const isEqualValue = isObjectValue
-        ? get(cachedOption, props.valueKey) === get(value, props.valueKey)
+        ? get(cachedOption.value, props.valueKey) === get(value, props.valueKey)
```

Comment:
```suggestion
        ? get(
            isObject(cachedOption.value) ? cachedOption.value : cachedOption, 
            props.valueKey
          ) === get(value, props.valueKey)
```

Reply:
I don't understand why doing this check. It only bring complex and confuse.

The struct should be same.

Reply:
I seem to have misunderstood your intention before, you mean that if `v-model` is object, `option.value` is also object...

Reply:
If so, maybe it should be added to the documentation, what do you think? @JeremyWuuuuu

Reply:
I think this is the behavior before #6933. Is any reason (issue or something else) for #6933?

Reply:
It has never been stated that `option.value` must be consistent with the `v-model` type, so if `v-model` is set to object, it needs to take effect.

Reply:
You can take a look at this test
https://github.com/element-plus/element-plus/blob/99848c1623fe8a24c0b8408b29c5868e66c83e05/packages/components/select/__tests__/select.spec.ts#L364

Reply:
This test is added in #6933

Reply:
I know, this test is the reason that created the pr #6933

Reply:
you can only choose one logic.

```
value-key = "c"
option = [{ value: "a", "label": "a"}]
v-model = {}

---
v-model = { "c" : "a" }

or 

v-model = "a"
```

```
value-key = "c"
option = [{ value: { "c": "a" }, "label": "a"}]
v-model = {}

---
v-model = undefined

or 

v-model = { "c": "a" }
```

Reply:
But if the team think compatible (little confusion) is a better solution, I'm ok with it.

Reply:
any update on this?

Reply:
WDYT @JeremyWuuuuu

Reply:
this is take too long. I think I will accept the suggestion code directly tomorrow.